dotnet-authserver/tests/TeacherIdentity.AuthServer.Tests/TestClients.cs

```diff
@@ -33,6 +33,39 @@ public static class TestClients
             $"{Permissions.Prefixes.Scope}{CustomScopes.UserWrite}",
 #pragma warning disable CS0618 // Type or member is obsolete
             $"{Permissions.Prefixes.Scope}{CustomScopes.Trn}",
+#pragma warning restore CS0618 // Type or member is obsolete
+        },
+        Requirements =
+        {
+            Requirements.Features.ProofKeyForCodeExchange
+        }
+    };
+
+    public static TeacherIdentityApplicationDescriptor ApplyForQts { get; } = new TeacherIdentityApplicationDescriptor()
+    {
+        ClientId = "apply-for-qts",
+        ClientSecret = "secret",
+        ConsentType = ConsentTypes.Implicit,
+        DisplayName = "Sample TeacherIdentity.TestClient app",
+        RedirectUris =
+        {
+            new Uri("https://localhost:1234/oidc/callback")
+        },
+        Permissions =
+        {
+            Permissions.Endpoints.Authorization,
+            Permissions.Endpoints.Token,
+            Permissions.GrantTypes.AuthorizationCode,
+            Permissions.GrantTypes.Password,
+            Permissions.ResponseTypes.Code,
+            Permissions.Scopes.Email,
+            Permissions.Scopes.Profile,
+            $"{Permissions.Prefixes.Scope}{CustomScopes.DqtRead}",
+            $"{Permissions.Prefixes.Scope}{CustomScopes.GetAnIdentitySupport}",
+            $"{Permissions.Prefixes.Scope}{CustomScopes.UserRead}",
+            $"{Permissions.Prefixes.Scope}{CustomScopes.UserWrite}",
+#pragma warning disable CS0618 // Type or member is obsolete
+            $"{Permissions.Prefixes.Scope}{CustomScopes.Trn}",
```

Comment:
Some of these custom scopes won't be used by AfQts so this would make it a little more production-like:
```suggestion
        Permissions =
        {
            Permissions.Endpoints.Authorization,
            Permissions.Endpoints.Token,
            Permissions.GrantTypes.AuthorizationCode,
            Permissions.ResponseTypes.Code,
            Permissions.Scopes.Email,
            Permissions.Scopes.Profile,
```